platforms/core-runtime/logging/src/main/java/org/gradle/internal/deprecation/DeprecationMessageBuilder.java

```diff
@@ -343,6 +381,13 @@ String formatAdvice(List<String> replacements) {
         }
     }
 
+    public static String createDefaultDeprecationIdString(String... ids) {
+        return screamingSnakeToKebabCase(StringUtils.join(ids, "-"))
+            .replaceAll("[^a-z-]", "-")
+            .replaceAll("-+", "-")
+            .replaceAll("^-+|-+$", "");
```

Comment:
I think we need some description here what is the expected input/output here and/or why we need regex here.